src/components/segmented-control/segmented-control.jsx

```diff
@@ -5,46 +5,104 @@ import SegmentedControlStyles from './segmented-control-styles';
 class SegmentedControl extends React.PureComponent {
   constructor(props, context) {
     super(props, context);
-    this.state = { selected: props.value };
+    if (this.props.type === 'radio') {
+      this.state = {
+        selected: props.value,
+      };
+    } else if (this.props.type === 'checkbox') {
+      // If multiple children
+      if (this.props.children.reduce) {
```

Comment:
Can do `Array.isArray(this.props.children)` if you want to make sure this is an array :)